internal/providers/msentraid/msentraid.go

```diff
@@ -286,7 +287,7 @@ func (p Provider) CurrentAuthenticationModesOffered(
 	log.Debugf(context.Background(), "In CurrentAuthenticationModesOffered: sessionMode=%q, supportedAuthModes=%q, tokenExists=%t, providerReachable=%t, endpoints=%q, currentAuthStep=%d\n", sessionMode, supportedAuthModes, tokenExists, providerReachable, endpoints, currentAuthStep)
 	var offeredModes []string
 	switch sessionMode {
-	case "passwd":
+	case sessionmode.ChangePassword, sessionmode.ChangePasswordOld:
```

Comment:
see, I find is weird that we have the helper `IsChangePaswordMode`, but still duplicate in the case the need to have both here an in noprovider.

Maybe we should just drop `IsChangePasswordMode` and have the 2 cases listed in `broker.go` too? That way, the day we clean it up would not be only "this is the only place to change", but "I need to look at every places with `ChangePasswordOld`.

(Ofc, you can object that we will remove the const, and so, have to replace the cases there too, but it would mean remove `IsChangePasswordMode` and reintroducing the equality too).

Reply:
done

Reply:
thx!